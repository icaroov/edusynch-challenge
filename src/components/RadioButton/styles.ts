import styled, { css } from 'styled-components'
import { darken } from 'polished'

export const Container = styled.div`
  ${({ theme }) => css`
    span {
      font-size: ${theme.font.sizes.small};
    }

    @media ${theme.breakpoints.tabletMax} {
      display: flex;
      flex-direction: column;
    }
  `}
`

export const Label = styled.span`
  ${({ theme }) => css`
    position: relative;
    display: flex;
    align-items: center;
    padding: 1rem 2rem;

    &::before {
      content: '';
      height: 1.8rem;
      width: 1.8rem;
      border: 3px solid ${theme.colors.secondary};
      border-radius: 50%;
      margin-right: 20px;
    }

    @media ${theme.breakpoints.tabletMax} {
      padding: 1rem;
    }
  `}
`

type RadioBoxProps = {
  isActive: boolean
}

export const RadioBox = styled.button<RadioBoxProps>`
  ${({ theme, isActive }) => css`
    display: flex;
    align-items: center;

    border: 2px solid
      ${isActive ? darken(0.2, theme.colors.white) : 'transparent'};
    padding: 1rem 2rem;
    border-radius: 4px;
    background-color: ${isActive
      ? theme.colors.primary
      : 'rgba(121, 85, 232, 0.3)'};

    color: ${theme.colors.white};
    font-weight: ${theme.font.bold};

    transition: 0.3s;

    &:hover {
      background-color: ${darken(0.05, theme.colors.primary)};
    }

    ${Label} {
      &::before {
        background-color: ${isActive
          ? theme.colors.secondary
          : darken(0.1, theme.colors.white)};
        border-color: ${isActive
          ? darken(0.4, theme.colors.tertiary)
          : 'transparent'};
      }
    }

    cursor: pointer;
  `}
`
