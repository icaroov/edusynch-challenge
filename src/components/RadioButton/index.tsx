import * as Styled from './styles'

export type RadioType = 'teacher' | 'student'

export type RadioButtonProps = {
  text: string
  typeRadio?: 'teacher' | 'student'
  isActive: boolean
  onCheck: () => void
}

const RadioButton = ({ text, isActive, onCheck }: RadioButtonProps) => {
  return (
    <Styled.Container>
      <Styled.RadioBox type="button" isActive={isActive} onClick={onCheck}>
        <Styled.Label>{text}</Styled.Label>
      </Styled.RadioBox>
    </Styled.Container>
  )
}

export default RadioButton
