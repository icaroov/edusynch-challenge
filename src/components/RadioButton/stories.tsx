import { useState } from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import RadioButton, { RadioButtonProps } from '.'

export default {
  title: 'RadioButton',
  args: {
    text: "I'M A TEACHER"
  },
  component: RadioButton
} as Meta

export const Default: Story<RadioButtonProps> = () => {
  const [checked, setChecked] = useState(false)

  return (
    <RadioButton
      isActive={checked}
      onCheck={() => setChecked(!checked)}
      text="DEFAULT"
    />
  )
}
