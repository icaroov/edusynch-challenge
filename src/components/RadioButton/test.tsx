import { screen } from '@testing-library/react'

import { renderWithTheme } from 'utils/tests/helpers'

import RadioButton from '.'

describe('<RadioButton />', () => {
  const onCheck = jest.fn()

  it('should render the Radio Button', () => {
    const { container } = renderWithTheme(
      <RadioButton isActive={false} onCheck={onCheck} text="Teste" />
    )

    expect(screen.getByRole('button', { name: /Teste/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the active button correctly', async () => {
    const onCheck = jest.fn()

    renderWithTheme(<RadioButton text="Teacher" onCheck={onCheck} isActive />)

    expect(screen.getByRole('button', { name: /Teacher/i })).toHaveStyle({
      'background-color': '#7A57FD'
    })
  })
})
