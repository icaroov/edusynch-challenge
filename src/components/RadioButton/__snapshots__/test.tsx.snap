// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> should render the Radio Button 1`] = `
.c0 span {
  font-size: 1.4rem;
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem 2rem;
}

.c3::before {
  content: '';
  height: 1.8rem;
  width: 1.8rem;
  border: 3px solid #FFD74F;
  border-radius: 50%;
  margin-right: 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 2px solid transparent;
  padding: 1rem 2rem;
  border-radius: 4px;
  background-color: rgba(121,85,232,0.3);
  color: #FFFFFF;
  font-weight: 700;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
}

.c1:hover {
  background-color: #663efd;
}

.c1 .c2::before {
  background-color: #e6e6e6;
  border-color: transparent;
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:768px) {
  .c3 {
    padding: 1rem;
  }
}

<div
  class="c0"
>
  <button
    class="c1"
    type="button"
  >
    <span
      class="c2 c3"
    >
      Teste
    </span>
  </button>
</div>
`;
